flake.nix

```diff
@@ -0,0 +1,76 @@
+{
+  description = "Tiny Tapeout tools packaged using poetry2nix";
+
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    poetry2nix = {
+      url = "github:nix-community/poetry2nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        # see https://github.com/nix-community/poetry2nix/tree/master#api for more functions and examples.
+        pkgs = nixpkgs.legacyPackages.${system};
+        p2n = poetry2nix.lib.mkPoetry2Nix { inherit pkgs; };
+        commit = if self ? "rev" then builtins.substring 0 7 self.rev else "dirty";
+        version = "6.0.0+${commit}";
+      in
+      {
+        packages = {
+          tt-tools = (p2n.mkPoetryApplication {
+            projectDir = self;
+
+            overrides = p2n.overrides.withDefaults (self: super: {
+
+              gdstk = super.gdstk.overridePythonAttrs (old: {
+                buildInputs = (old.buildInputs or [ ]) ++ [ self.setuptools pkgs.zlib pkgs.qhull ];
+                nativeBuildInputs = (old.nativeBuildInputs or [ ]) ++ [ pkgs.cmake ];
+                dontUseCmakeConfigure = true;
+              });
+
+              defusedxml = super.defusedxml.overridePythonAttrs (old: {
+                buildInputs = (old.buildInputs or [ ]) ++ [ self.setuptools ];
+              });
+
+              pillow = super.pillow.overridePythonAttrs (old: {
+                # Use preConfigure from nixpkgs to fix library detection issues and
+                # impurities which can break the build process; this also requires
+                # adding propagatedBuildInputs and buildInputs from the same source.
+                propagatedBuildInputs = (old.buildInputs or [ ]) ++ [ self.olefile self.defusedxml ];
+                buildInputs = (old.buildInputs or [ ]) ++ pkgs.python3.pkgs.pillow.buildInputs;
+                preConfigure = (old.preConfigure or "") + pkgs.python3.pkgs.pillow.preConfigure;
+
+                # https://github.com/nix-community/poetry2nix/issues/1139
+                patches = (old.patches or [ ]) ++ pkgs.lib.optionals (old.version == "9.5.0") [
+                  (pkgs.fetchpatch {
+                    url = "https://github.com/python-pillow/Pillow/commit/0ec0a89ead648793812e11739e2a5d70738c6be5.diff";
+                    sha256 = "sha256-rZfk+OXZU6xBpoumIW30E80gRsox/Goa3hMDxBUkTY0=";
+                  })
+                ];
+              });
+
+            });
+          }).overridePythonAttrs (old:
+            {
+              inherit version;
+              patchPhase = ''
+                sed \
+                  -e 's/tt_version\s*=.*/tt_version = "${version}"/' \
+                  -e 's|yowasp-yosys|${pkgs.yosys}/bin/yosys|' \
```

Comment:
For some reason, yowasp-yosys is locking in/installing the windows binary:
```
  File "/nix/store/5gpl9xnv0zxj0wxrcv6l0n72zhg89235-python3.11-wasmtime-9.0.0/lib/python3.11/site-packages/wasmtime/_ffi.py", line 30, in <module>
    raise RuntimeError("precompiled wasmtime binary not found at `{}`".format(filename))
RuntimeError: precompiled wasmtime binary not found at `/nix/store/5gpl9xnv0zxj0wxrcv6l0n72zhg89235-python3.11-wasmtime-9.0.0/lib/python3.11/site-packages/wasmtime/linux-x86_64/_libwasmtime.so`
```
but
```
❯ ls /nix/store/5gpl9xnv0zxj0wxrcv6l0n72zhg89235-python3.11-wasmtime-9.0.0/lib/python3.11/site-packages/wasmtime/
__init__.py  _bindings.py  _engine.py  _exportable.py  _ffi.py   _globals.py   _linker.py  _module.py  _table.py  _types.py  _wasi.py      bindgen    py.typed
__pycache__  _config.py    _error.py   _extern.py      _func.py  _instance.py  _memory.py  _store.py   _trap.py   _value.py  _wat2wasm.py  loader.py  win32-x86_64
```
has `win32-x86_64` not `linux-x86_64` -- and given Nix has yosys thought it's easier to just use that rather than yowasp-yosys.
```